kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -280,6 +280,31 @@ ExpressionTransformResult visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ExpressionTransformResult visitLike(final Predicate like) {
+            List<ExpressionTransformResult> children =
+                    like.getChildren().stream()
+                        .map(this::visit)
+                        .collect(Collectors.toList());
+            int size = children.size();
+            if (size<2 || size>3) {
+                throw unsupportedExpressionException(like,
+                        "like requires at least two expressions and a maximum of 3 expressions");
+            }
+
+            Predicate transformedExpression =
+                    LikeExpressionEvaluator.validateAndTransform(
+                        like,
+                        children.get(0).expression,
+                        children.get(0).outputType,
+                        children.get(1).expression,
+                        children.get(1).outputType,
+                        size==3 ? children.get(2).expression : null,
+                        size==3 ? children.get(2).outputType : null);
```

Comment:
can pass the child expression and types separately, that way few arguments for the method?

```
Predicate transformedExpression = LikeExpressionEvaluator.validateAndTransform(
                    like,
                    children.stream().map(e -> e.expression).collect(Collectors.toList()),
                    children.stream().map(e -> e.outputType).collect(Collectors.toList()));
```